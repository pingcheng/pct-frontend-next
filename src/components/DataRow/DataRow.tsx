import { ReactNode } from "react";

type DataRowProps = {
  label: string;
  children?: ReactNode;
};

export function DataRow({ label, children }: DataRowProps) {
  return (
    <div className="flex flex-col sm:flex-row items-start sm:items-baseline mb-6 pb-4 border-b border-dashed border-gray-200 dark:border-gray-700 last:border-b-0 last:mb-0 last:pb-0">
      <div className="text-gray-500 dark:text-gray-400 w-full sm:w-1/4 text-left sm:text-right pr-2 font-semibold tracking-wide uppercase text-xs mb-1 sm:mb-0 pt-0 sm:pt-1">
        {label.trim()}
      </div>
      <div className="flex-1 pl-0 sm:pl-2 w-full">{children}</div>
    </div>
  );
}
