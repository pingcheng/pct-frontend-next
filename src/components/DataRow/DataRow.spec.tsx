import { render, screen } from "@testing-library/react";
import { DataRow } from "@/components/DataRow/DataRow";

describe("test <DataRow />", () => {
  const defaultLabel = "test label";
  const defaultValue = "This is default value";

  it("should match snapshot", () => {
    const { container } = render(
      <DataRow label={defaultLabel}>{defaultValue}</DataRow>,
    );
    expect(container).toMatchSnapshot();
  });

  it("should contain label as all upper case", () => {
    render(<DataRow label={defaultLabel}>{defaultValue}</DataRow>);
    expect(screen.getByText(defaultLabel.toUpperCase())).toBeInTheDocument();
  });

  it("should contain value as text", () => {
    render(<DataRow label={defaultLabel}>{defaultValue}</DataRow>);
    expect(screen.getByText(defaultValue)).toBeInTheDocument();
  });
});
