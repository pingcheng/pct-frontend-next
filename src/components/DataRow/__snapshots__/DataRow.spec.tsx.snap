// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`test <DataRow /> should match snapshot 1`] = `
<div>
  <div
    class="flex flex-col sm:flex-row items-start sm:items-baseline py-3 border-b border-dashed border-gray-200 dark:border-gray-700 last:border-b-0 last:pb-0"
  >
    <div
      class="text-gray-500 dark:text-gray-400 w-full sm:w-1/4 text-left sm:text-right pr-2 sm:pr-3 font-semibold tracking-wide uppercase text-xs pt-0 sm:pt-1"
    >
      test label
    </div>
    <div
      class="flex-1 pl-0 sm:pl-3 w-full"
    >
      This is default value
    </div>
  </div>
</div>
`;
